inference-engine/src/vpu/common/src/ngraph/transformations/merge_subsequent_dsr_operations.cpp

```diff
@@ -20,7 +20,9 @@ MergeSubsequentDSROperations::MergeSubsequentDSROperations() {
         if (!predecessor) {
             return false;
         }
-
+        for (auto& consumer : predecessor->get_output_target_inputs(0)) {
+            consumer.replace_source_output(dsr);
```

Comment:
This is a very interesting case! You have DSR1->DSR2 sequence and you want to reconnect all DSR1 consumers to DSR2 and finally remove DSR1. But in your solution you get a cyclic dependency because DSR1 consumers contain DSR2 operation so DSR2 will be consumed by itself after replacement. 
I would suggest to implement such replacement in this way (this is just pseudocode):
```cpp
// this will create a new DSR with correct inputs
auto new_dsr = dsr->copy_with_new_inputs(predprocessor->input_value(0), dsr->intput_value(1));
// replace DSR2 with new so DSR2 will lose all consumers so it will die after pass execution
replace_node(dsr2, new_dsr);
// reconnect all DSR1 consumers even with DSR2 which will be destructed so this is no more an issue
for (auto& consumer : predecessor->get_output_target_inputs(0)) {
            consumer.replace_source_output(new_dsr);
}
```

Reply:
Thanks a lot for a good comment. Fixed now.

Reply:
Why DSR1 consumers contain DSR2 operation?
I understand why DSR2 will be consumed by itself, but isn't it easier just to add a condition inside for loop?

Reply:
DSR1 has DSR2 as a consumer because it is directly connected with it.
We can handle this situation in multiple ways but the way that I proposed is much more natural to nGraph and doesn't contain any extra logic.

Reply:
Ok, I got it. Thanks for explanation!
Just a loop with if inside still looks simpler to me than 2 extra calls/logic as copy with new inputs and replace.

@KorA6 I don't insist, it's up to you